src/core/qgsvectorlayerlabeling.cpp

```diff
@@ -88,3 +92,469 @@ QgsVectorLayerSimpleLabeling *QgsVectorLayerSimpleLabeling::create( const QDomEl
+      scaleProps.insert( "scaleMaxDenom", qgsDoubleToString( mSettings->maximumScale ) );
+      QgsSymbolLayerUtils::applyScaleDependency( doc, ruleElement, scaleProps );
+    }
+
+    // text symbolizer
+    QDomElement textSymbolizerElement = doc.createElement( QStringLiteral( "se:TextSymbolizer" ) );
+    ruleElement.appendChild( textSymbolizerElement );
+
+    // label
+    QgsTextFormat format = mSettings->format();
+    QFont font = format.font();
+    QDomElement labelElement = doc.createElement( QStringLiteral( "se:Label" ) );
+    textSymbolizerElement.appendChild( labelElement );
+    if ( mSettings->isExpression )
+    {
+      labelElement.appendChild( doc.createComment( QStringLiteral( "SE Export for %1 not implemented yet" ).arg( mSettings->getLabelExpression()->dump() ) ) );
+      labelElement.appendChild( doc.createTextNode( "Placeholder" ) );
+    }
+    else
+    {
+      if ( font.capitalization() == QFont::AllUppercase )
+      {
+        appendSimpleFunction( doc, labelElement, QStringLiteral( "strToUpperCase" ), mSettings->fieldName );
+      }
+      else if ( font.capitalization() == QFont::AllLowercase )
+      {
+        appendSimpleFunction( doc, labelElement, QStringLiteral( "strToLowerCase" ), mSettings->fieldName );
+      }
+      else if ( font.capitalization() == QFont::Capitalize )
+      {
+        appendSimpleFunction( doc, labelElement, QStringLiteral( "strCapitalize" ), mSettings->fieldName );
+      }
+      else
+      {
+        QDomElement propertyNameElement = doc.createElement( QStringLiteral( "ogc:PropertyName" ) );
+        propertyNameElement.appendChild( doc.createTextNode( mSettings->fieldName ) );
+        labelElement.appendChild( propertyNameElement );
+      }
+    }
+
+    // font
+    QDomElement fontElement = doc.createElement( QStringLiteral( "se:Font" ) );
+    textSymbolizerElement.appendChild( fontElement );
+    fontElement.appendChild( QgsSymbolLayerUtils::createSvgParameterElement( doc, QStringLiteral( "font-family" ), font.family() ) );
+    double fontSize = QgsSymbolLayerUtils::rescaleUom( format.size(), format.sizeUnit(), props );
+    fontElement.appendChild( QgsSymbolLayerUtils::createSvgParameterElement( doc, QStringLiteral( "font-size" ), QString::number( fontSize ) ) );
+    if ( format.font().italic() )
+    {
+      fontElement.appendChild( QgsSymbolLayerUtils::createSvgParameterElement( doc, QStringLiteral( "font-style" ), QStringLiteral( "italic" ) ) );
+    }
+    if ( format.font().bold() )
+    {
+      fontElement.appendChild( QgsSymbolLayerUtils::createSvgParameterElement( doc, QStringLiteral( "font-weight" ), QStringLiteral( "bold" ) ) );
+    }
+
+    // label placement
+    QDomElement labelPlacement = doc.createElement( QStringLiteral( "se:LabelPlacement" ) );
+    textSymbolizerElement.appendChild( labelPlacement );
+    double maxDisplacement = 0;
+    double repeatDistance = 0;
+    if ( mSettings->placement == QgsPalLayerSettings::OverPoint )
```

Comment:
Keeping with the general theme of my review - these if blocks should be switches instead ;)